transcom/mymove
# B 20289 int

## [B-20289](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-20289)

## Summary
This story changes the weight fields on the SSW to reflect the final values in the payment packet only. In order to support this, a flag was created to differentiate between the AOA and Payment Packets upon creation, so that different pieces of data may be filled into the form depending on which document is to be generated.

## Verification Steps for the Author

These are to be checked by the author.


## Verification Steps for Reviewers

These are to be checked by a reviewer.


### How to test

1. Access the app as a customer. We'll have to make a new move and run through the entire PPM process, as our test data may not have everything we need for generating these documents.
2. Create a new customer, move, and PPM Shipment. When creating the PPM Shipment, ensure an advance is requested.
3. Access the office app as a Service Counselor. Review the orders and PPM info. Approve the advance and the PPM. Download the AOA Packet. Ensure that the estimated weight is displayed in the AOA Packet in the fields in the screenshot below, as well as the shipment weight field on the second page.
4. Log back in as the customer, and submit your move information. When submitting your weight ticket, ensure that the full and empty load difference is **different** than the estimated shipment weight. Submit your moving expenses.
5. Log back in as the service counselor, and review the moving expenses. Approve them, complete the closeout process, and download the Payment Packet.
6. Ensure that the Payment Packet lists the final PPM weight in the same fields where the Estimated weight appeared in the AOA Packet.
7. Ensure all server tests pass.

### Backend


## Screenshots
AOA:
<img width="294" alt="image" src="https://github.com/transcom/mymove/assets/148906585/84cbf105-9254-49f9-b86a-19b3d56d2562">
Payment Packet:
<img width="294" alt="image" src="https://github.com/transcom/mymove/assets/148906585/ee2a59fc-61a2-40e0-b0e9-d2bec699a73f">
